src/main/java/seedu/duke/model/gates/Gate.java

```diff
@@ -29,4 +61,34 @@ public int getInput() {
     public boolean isSetInput() {
         return isSetInput;
     }
+
+    /**
+     * Returns whether second input is set.
+     *
+     * @return boolean
+     */
+    public boolean isSetSecondInput() {
+        return isSetSecondInput;
+    }
+
+    /**
+     * Returns if both inputs are set or not.
+     *
+     * @return boolean value
+     */
+    public boolean hasSetBothInput() {
```

Comment:
Why is this necessary? Doesn't seem to be used outside of `checkHasSetBothInput()`.

Reply:
You're right! And I was thinking the same, just kept it in case it would be used somewhere else...should I remove it?

Reply:
Yep can remove.